src/main/java/com/google/devtools/build/lib/runtime/commands/TargetPatternFileSupport.java

```diff
@@ -0,0 +1,50 @@
+package com.google.devtools.build.lib.runtime.commands;
+
+import com.google.common.collect.Lists;
+import com.google.devtools.build.lib.buildtool.BuildRequestOptions;
+import com.google.devtools.build.lib.profiler.Profiler;
+import com.google.devtools.build.lib.profiler.SilentCloseable;
+import com.google.devtools.build.lib.runtime.CommandEnvironment;
+import com.google.devtools.build.lib.vfs.FileSystemUtils;
+import com.google.devtools.build.lib.vfs.Path;
+import com.google.devtools.common.options.OptionsParsingResult;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+public final class TargetPatternFileSupport {
```

Comment:
make package private (remove "public")? Also add a quick javadoc to this and the visible methods and exception.